relayer/tests/browser_test.ts

```diff
@@ -0,0 +1,229 @@
+    // should expect to throw
+    expect(() => {
+      AnchorProvider.local("http://127.0.0.1:8899", confirmConfig);
+    }).to.throw("Provider local is not available on browser.");
+  });
+
+  it("(browser) should shield and update merkle tree", async () => {
+    let testInputs = {
+      amountSol: 15,
+      token: "SOL",
+      type: Action.SHIELD,
+      expectedUtxoHistoryLength: 1,
+    };
+    const userTestAssertHelper = new UserTestAssertHelper({
+      userSender: user,
+      userRecipient: user,
+      provider,
+      testInputs,
+    });
+    await userTestAssertHelper.fetchAndSaveState();
+
+    await user.shield({
+      publicAmountSol: testInputs.amountSol,
+      token: testInputs.token,
+      confirmOptions: ConfirmOptions.spendable,
+    });
+
+    await waitForBalanceUpdate(userTestAssertHelper, user);
+    await userTestAssertHelper.checkSolShielded();
+  });
+
+  it("(browser) should unshield SOL and update merkle tree", async () => {
+    const solRecipient = Keypair.generate();
+
+    const testInputs = {
+      amountSol: 1,
+      token: "SOL",
+      type: Action.UNSHIELD,
+      recipient: solRecipient.publicKey,
+      expectedUtxoHistoryLength: 1,
+    };
+
+    const userTestAssertHelper = new UserTestAssertHelper({
+      userSender: user,
+      userRecipient: user,
+      provider,
+      testInputs,
+    });
+    // need to wait for balance update to fetch current utxos
+    await user.getBalance();
+    await userTestAssertHelper.fetchAndSaveState();
+    await user.unshield({
+      publicAmountSol: testInputs.amountSol,
+      token: testInputs.token,
+      recipient: testInputs.recipient,
+      confirmOptions: ConfirmOptions.spendable,
+    });
+
+    await waitForBalanceUpdate(userTestAssertHelper, user);
+    await userTestAssertHelper.checkSolUnshielded();
+  });
+
+  it("should transfer sol and update merkle tree ", async () => {
+    const testInputs = {
+      amountSol: 1,
+      token: "SOL",
+      type: Action.TRANSFER,
+      expectedUtxoHistoryLength: 1,
+      recipientSeed: bs58.encode(new Uint8Array(32).fill(9)),
+      expectedRecipientUtxoLength: 1,
+    };
+
+    const recipientAccount = new Account({
+      poseidon,
+      seed: testInputs.recipientSeed,
+    });
+
+    const userRecipient: User = await User.init({
+      provider,
+      seed: testInputs.recipientSeed,
+    });
+
+    const testStateValidator = new UserTestAssertHelper({
+      userSender: user,
+      userRecipient,
+      provider,
+      testInputs,
+    });
+    await testStateValidator.fetchAndSaveState();
+    // need to wait for balance update to fetch current utxos
+    await user.getBalance();
+    await user.transfer({
+      amountSol: testInputs.amountSol,
+      token: testInputs.token,
+      recipient: recipientAccount.getPublicKey(),
+    });
+
+    // await waitForBalanceUpdate(testStateValidator, user);
+    await sleep(6000);
```

Comment:
What's the rationale for using sleep over waitForBalanceUpdate?
(Please remove one or the other, although I think waitForBalanceUpdate is more robust)